### PR TITLE
Database Controller

### PR DESCRIPTION
**Pull request by mprtcz**
Added a db controller which will save users in the db, load them and update their score if necessary.
Also added tests that work on the in-memory database as well as a travis config file which specifies remote CI execution.

### CODE REVIEW

File: src/main/java/com/spanish_inquisition/battleship/server/database/DatabaseController.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    private static void loadJDBCDriver() {
+        try {
+            Class.forName(JDBC_DRIVER);
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void savePlayer(PlayerEntity playerEntity) {
+        loadJDBCDriver();
```
Comment (Devas) [inline src/main/java/com/spanish_inquisition/battleship/server/database/DatabaseController.java]: Why do you load multiple times? Do it once in constructor

File: src/main/java/com/spanish_inquisition/battleship/server/database/DatabaseController.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        try (Connection connection = DriverManager.getConnection(DATABASE);
+             Statement statement = connection.createStatement()) {
+            createTableIfNotExists(statement);
+            PreparedStatement preparedStatement = connection.prepareStatement(String.format("insert into %s (name, score) values (?, ?);", TABLE_NAME));
+            setSaveStatementStrings(preparedStatement, playerEntity);
+            preparedStatement.executeBatch();
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (Devas) [inline src/main/java/com/spanish_inquisition/battleship/server/database/DatabaseController.java]: Use logger

File: src/test/java/com/spanish_inquisition/battleship/server/database/DatabaseControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+public class DatabaseControllerTest {
+    private DatabaseController databaseController;
+    private static final String DB_NAME = "history-test.db";
+    private static final String TABLE_NAME = DatabaseController.TABLE_NAME;
+
+    @BeforeClass
+    public void setUpDBController() {
+        databaseController = new DatabaseController();
+        DatabaseController.JDBC_DRIVER = "org.h2.Driver";
+        DatabaseController.DATABASE = "jdbc:h2:./dbs/" +DB_NAME;
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/server/database/DatabaseControllerTest.java]: Reformat code always

File: src/test/java/com/spanish_inquisition/battleship/server/database/DatabaseControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    @AfterMethod
+    private void dropAfterMethod() {
+        try (Connection connection = DriverManager.getConnection(DatabaseController.DATABASE);
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/server/database/DatabaseControllerTest.java:44]: 👍

File: src/test/java/com/spanish_inquisition/battleship/server/database/DatabaseControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        assertEquals(entities.size(), 1);
+    }
+
+    @Test
+    public void testLoadPlayers() throws Exception {
+        insertTwoPlayersIntoDB();
+        List<PlayerEntity> entities = databaseController.loadPlayers();
+        System.out.println("entities = " + entities);
+        assertEquals(entities.size(), 2);
+        assertTrue(entities.contains(new PlayerEntity("test1", 1)));
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/server/database/DatabaseControllerTest.java:77]: Still not using SoftAssert

**Review by Devas — APPROVED** [review]

File: src/main/java/com/spanish_inquisition/battleship/server/database/DatabaseController.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        }
+    }
+
+    private void updatePlayer(PlayerEntity playerEntity) {
+        loadJDBCDriver();
+        try (Connection connection = DriverManager.getConnection(DATABASE);
+             Statement statement = connection.createStatement()) {
+            createTableIfNotExists(statement);
```
Comment (patkkka) [inline src/main/java/com/spanish_inquisition/battleship/server/database/DatabaseController.java]: You don't have to execute this here - if updatePlayer is called - you are sure that table exists.

**Review by patkkka — APPROVED** [review]

**Pull request merged**